### PR TITLE
Add blogging prompt list item view and adapter

### PR DESCRIPTION
**Pull request by metalwihen**
Fixes #17125 

This PR introduces the UI for the Prompts List Items.

**To test:**
- Follow test instructions as explained in previous PRs: #2, #5, #6
- Verify that the UI of the prompt list items matches the requirements

**Screenshot:**
| Loading | Loaded |
| ----------- | ----------- |
| image |  image |

## Regression Notes
1. Potential unintended areas of impact
- Purely UI change. Should not impact any other class.

2. What I did to test those areas of impact (or what existing automated tests I relied on)
- Manual testing around the Prompt List flow. 

3. What automated tests I added (or what prevented me from doing so)
- None, since this was a UI change.

PR submission checklist:

- [x] I have completed the Regression Notes.
- [x] I have considered adding accessibility improvements for my changes.
- [x] I have considered if this change warrants user-facing release notes and have added them to `RELEASE-NOTES.txt` if necessary.

### CODE REVIEW

File: WordPress/src/main/java/org/wordpress/android/ui/bloggingprompts/BloggingPromptsListAdapter.kt [hunk 1]
```diff
@@ -0,0 +1,77 @@
...
+}
+
+class BloggingPromptsListItemViewHolder(
+    private val binding: BloggingPromptsListItemBinding
+) : RecyclerView.ViewHolder(binding.root) {
+    fun bind(prompt: BloggingPromptsListItem) {
+        with(binding) {
+            promptTitle.text = prompt.title
+            promptSubtitleAnswerCount.text =
+                    when (prompt.respondentsCount) {
```
Comment (RenanLukas) [inline WordPress/src/main/java/org/wordpress/android/ui/bloggingprompts/BloggingPromptsListAdapter.kt]: Since we can't use plurals in the project, do you think it would make sense to extract this into an extension function, so it can be reused? Something like:

```
    fun View.getQuantityString(
        quantity: Int,
        @StringRes zeroRes: Int,
        @StringRes oneRes: Int,
        @StringRes otherRes: Int
    ) = when (quantity) {
        0 -> context.getString(zeroRes)
        1 -> context.getString(oneRes)
        else -> context.getString(otherRes)
    }
```

File: WordPress/src/main/java/org/wordpress/android/ui/bloggingprompts/BloggingPromptsListAdapter.kt [hunk 2]
```diff
@@ -0,0 +1,77 @@
...
+                        0 -> binding.root.context.getString(R.string.blogging_prompts_list_item_count_answers_zero)
+                        1 -> binding.root.context.getString(R.string.blogging_prompts_list_item_count_answers_one)
+                        else -> binding.root.context.getString(
+                                R.string.blogging_prompts_list_item_count_answers_many,
+                                prompt.respondentsCount
+                        )
+                    }
+            promptSubtitleDate.text =
+                    SimpleDateFormat("MMM d", Locale.getDefault()).format(prompt.date)
```
Comment (RenanLukas) [inline WordPress/src/main/java/org/wordpress/android/ui/bloggingprompts/BloggingPromptsListAdapter.kt]: Would it make sense to extract this mapping to another class (e.g. ViewModel) instead so that it can be unit tested? In that case maybe we could use `Locale` from `LocaleManagerWrapper`.

Also, would it make sense to extract the date format to a constant?

File: WordPress/src/main/java/org/wordpress/android/ui/bloggingprompts/BloggingPromptsListFragment.kt [hunk 3]
```diff
@@ -26,31 +31,52 @@ class BloggingPromptsListFragment : ViewPagerFragment() {
...
                 PromptSection.ANSWERED -> showNoConnection()
             }.exhaustive
         }, TimeUnit.SECONDS.toMillis(2))
     }
 
-    private fun showContent() {
+    private fun initializeViews() {
         with(binding) {
-            tempText.setVisible(true)
+            promptsList.layoutManager = LinearLayoutManager(activity, RecyclerView.VERTICAL, false)
```
Comment (RenanLukas) [inline WordPress/src/main/java/org/wordpress/android/ui/bloggingprompts/BloggingPromptsListFragment.kt]: Would it make sense to use `apply` here? Something like:

```
promptsList.apply {
    layoutManager = ...
    addItemDecoration(...)
}
```

File: WordPress/src/main/java/org/wordpress/android/ui/bloggingprompts/BloggingPromptsListFragment.kt [hunk 4]
```diff
@@ -26,31 +31,52 @@ class BloggingPromptsListFragment : ViewPagerFragment() {
...
+            promptsList.addItemDecoration(DividerItemDecoration(context, LinearLayout.VERTICAL))
+            promptsListAdapter = BloggingPromptsListAdapter()
+            promptsList.adapter = promptsListAdapter
+        }
+    }
+
+    private fun showContent(list: List<BloggingPromptsListItem>) {
+        with(binding) {
+            promptsListAdapter.update(list)
+            promptsList.setVisible(true)
```
Comment (RenanLukas) [inline WordPress/src/main/java/org/wordpress/android/ui/bloggingprompts/BloggingPromptsListFragment.kt:73]: Would it make sense to have the flags of what should be visible or not in each state from the ViewModel so they can be unit tested?

**Review by RenanLukas — COMMENTED** [review]
Thanks for the PR @metalwihen 👍 
I've left a few comments/questions in my initial review. Please let me know if they make sense.

File: WordPress/src/main/java/org/wordpress/android/ui/bloggingprompts/BloggingPromptsListAdapter.kt (hunk 2 above)
Comment (metalwihen) [reply WordPress/src/main/java/org/wordpress/android/ui/bloggingprompts/BloggingPromptsListAdapter.kt]: Yes. Will do.

File: WordPress/src/main/java/org/wordpress/android/ui/bloggingprompts/BloggingPromptsListFragment.kt (hunk 3 above)
Comment (metalwihen) [reply WordPress/src/main/java/org/wordpress/android/ui/bloggingprompts/BloggingPromptsListFragment.kt]: Agreed. Will do.

File: WordPress/src/main/java/org/wordpress/android/ui/bloggingprompts/BloggingPromptsListAdapter.kt (hunk 1 above)
Comment (metalwihen) [reply WordPress/src/main/java/org/wordpress/android/ui/bloggingprompts/BloggingPromptsListAdapter.kt]: Makes sense. Will make it an extension method and move it to `ViewExtensions.kt`

File: WordPress/src/main/java/org/wordpress/android/ui/bloggingprompts/BloggingPromptsListFragment.kt (hunk 4 above)
Comment (metalwihen) [reply WordPress/src/main/java/org/wordpress/android/ui/bloggingprompts/BloggingPromptsListFragment.kt:73]: I don't believe this is necessary for the approach I've used because we're focusing on defining different states for the UI. The fragment (View) is responsible for rendering 5 different states (Loaded, Loading, Error, Empty and No_Connection). The presentation logic of **when** to show a state is tested via unit tests. The individual view elements and their visibility may not add much value in unit tests since each state is independent of the other (eg: Empty screen won't be shown when Loading screen or Loaded screen is shown). 

```
showEmpty()
```
vs 
```
showEmptyView() // shared with Loading|Error|NoConection|
hideContentView()
```

Comment (metalwihen) [reply WordPress/src/main/java/org/wordpress/android/ui/bloggingprompts/BloggingPromptsListFragment.kt:73]: That said, if you still feel it's better to unit-test the visibility flags as well, I can make the changes.

File: WordPress/src/main/java/org/wordpress/android/ui/bloggingprompts/BloggingPromptsListAdapter.kt (hunk 1 above)
Comment (metalwihen) [reply WordPress/src/main/java/org/wordpress/android/ui/bloggingprompts/BloggingPromptsListAdapter.kt]: This is done.

File: WordPress/src/main/java/org/wordpress/android/ui/bloggingprompts/BloggingPromptsListAdapter.kt (hunk 2 above)
Comment (metalwihen) [reply WordPress/src/main/java/org/wordpress/android/ui/bloggingprompts/BloggingPromptsListAdapter.kt]: This is done. 

Note - Since the ViewModel for the list fragment will be introduced in the next PR, I've moved the date logic to the dummy-provider block for now. This will be moved to the appropriate class when creating the ViewModel.

File: WordPress/src/main/java/org/wordpress/android/ui/bloggingprompts/BloggingPromptsListFragment.kt (hunk 3 above)
Comment (metalwihen) [reply WordPress/src/main/java/org/wordpress/android/ui/bloggingprompts/BloggingPromptsListFragment.kt]: This is done. I've used `with()` instead of `apply` though.

File: WordPress/src/main/java/org/wordpress/android/ui/bloggingprompts/BloggingPromptsListFragment.kt (hunk 4 above)
Comment (develric) [reply WordPress/src/main/java/org/wordpress/android/ui/bloggingprompts/BloggingPromptsListFragment.kt:73]: Hey @metalwihen 👋 , I understand we are still in a stage where there is some dummy code to populate the UI for poc purpose, this said I think I agree with Renan advice;  if I get your reasoning you are going to have in the ViewModel the logic of which of the different presentation function to trigger in the various point in time (and I like this 👍 ), regardless I usually think that the more we can take out pieces of logic from the View and have them covered by unit test the better (also simple logic like deciding if a setVisible on a view should be a true or a false). It should also be possible to leverage a more DRY approach by factoring common code that now repeats in the various showEmpty or showContent or similar functions (thinking as an example to promptsList.setVisible).

Wdyt?

**Comment by develric** [comment]
Thanks for the changes @metalwihen 🙇 . As you know I'm chiming in to side help this week since Renan is AFK, so sorry if what I'm going to say could sound redundant or already discussed 😄  . 

I tested the code and it works as expected. I have a couple of notes to get your thoughts on, even though since we have an upcoming PR (that you mentioned should also add the ViewModel) and afaiu we are working against a feature branch on your fork (correct?) I think it's good to have your feedback here, approve this PR and move to the next one; I guess we may start having real data populating the UI depending on the breakdown you were planning for upcoming work of course, let me know 🙇 . 

1. This is not specific to this PR but more to check if it was on the radar already. I think we should have a strategy (Pull To Refresh or a Retry button?) to allow the user to exit from error conditions (no network or the like). If it was not discussed, would you mind adding a comment about it in the [original issue](https://github.com/wordpress-mobile/WordPress-Android/issues/17125) so we don't forget? wdyt? 
2. I left a comment [here](https://github.com/metalwihen/WordPress-Android/pull/8#discussion_r997901710) I would like to have your thoughts on; again, since you mentioned the upcoming PR is adding the ViewModel, I think we can approve this one and in case tackle that in the upcoming PR(s), but happy to have your thoughts  

Let me know wdyt 🙇 

Side note: in general I think it's useful to keep separate commits for code that is addressing a specific feedback so it's easier to point the reviewer to the relevant code. Maybe something we could keep in mind for next PR, wdyt?

**Comment by metalwihen** [comment]
> sorry if what I'm going to say could sound redundant or already discussed

That's completely fine. ^_^ If there's anything you'd like to know, please ask. I appreciate you taking the time. 

> exit from error conditions

Agreed. A button makes sense. Will add it to next steps. 

>  I think it's good to have your feedback here, approve this PR and move to the next one;

In agreement. Let me note all the next steps here which can be picked up in the next PR:

**Next steps:**
* Add the network layer and relevant usecases
* Introduce the ViewModel with tests 
* Refactor the UI approach to handle visibility flags in viewmodel
* Add a button in the error|empty|no_connection screen to retry

>  I think it's useful to keep separate commits for code that is addressing a specific feedback

Noted. Thank you for the feedback. My original intent was to reduce noise with multiple commits. But I understand that separate commits would definitely make it easier for the reviewer.

File: WordPress/src/main/java/org/wordpress/android/ui/bloggingprompts/BloggingPromptsListFragment.kt (hunk 4 above)
Comment (metalwihen) [reply WordPress/src/main/java/org/wordpress/android/ui/bloggingprompts/BloggingPromptsListFragment.kt:73]: Thank you clarifying. :) I'll make the changes. I plan to do this in the next PR once the viewmodel is introduced. 

> leverage a more DRY approach by factoring common code

Will think more on this and get back to you.

**Review by develric — APPROVED** [review]
> Next steps:
> - Add the network layer and relevant usecases
> - Introduce the ViewModel with tests
> - Refactor the UI approach to handle visibility flags in viewmodel
> - Add a button in the error|empty|no_connection screen to retry

That seems correct 👍 ; let's stay with the button that seems pretty straightforward for the sake of poc in the context of an `ActionableEmptyView`, we can review it later with design where it makes sense 👍 

>next steps here which can be picked up in the next PR

I'll let you decide if separating in more than 1 PR makes sense or not 😊

:shipit:
